src/cargo/core/compiler/fingerprint/dep_info.rs

```diff
@@ -84,7 +84,7 @@ pub enum DepInfoPathType {
 ///   | len of key | key bytes | value exists? | len of value | value bytes |
```

Comment:
Should we also have a sanity-check value on "len of key"  We likely should never have a dep info key that is larger than 10k for instance

Reply:
(and ture for any length we deal with)

Reply:
Not sure. What if they do track that many environment variables?

An allocation error seems reasonable in that case. Cargo may crash in other places even it doesn't here.

Reply:
An alternative is <https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.try_with_capacity>.
(not yet stable though)